langlib/lang.value/src/main/ballerina/src/lang.value/value.bal

```diff
@@ -117,9 +117,26 @@ public function toString((any|error) v) returns string = external;
 
 # Returns the string that represents `v` in JSON format.
 #
-# + v - json value
+# + v - anydata value
 # + return - string representation of json
-public function toJsonString(json v) returns string = external;
+public function toJsonString(anydata v) returns string = external;
+
+# Converts a value of type `anydata` to `json`.
+# This does a deep copy of `v` converting values that do
+# not belong to json into values that do.
+# A value of type `xml` is converted into a string as if
```

Comment:
XML conversion doesn't seem to work when its in a map?

```ballerina
public function main() {
    xml x = xml `<foo><bar/>Hello World</foo>`;
    json j1 = x.toJson(); // WORKS

    map<anydata> m1 = {a: 1, b: "str"};
    json j2 = m1.toJson(); // WORKS

    map<anydata> m2 = {a: 1, b: x};
    json j3 = m2.toJson(); // ERROR - error: {ballerina/lang.typedesc}ConversionError 
                        // message='map<anydata>' value cannot be converted to 'json'
}
```

Reply:
Solved